src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorCSharpLanguageMiddleware.ts

```diff
@@ -69,10 +69,33 @@ export class RazorCSharpLanguageMiddleware implements LanguageMiddleware {
                         continue;
                     }
                 } else {
+                    // Similar to above, this is kind of wrong. We're manually trimming whitespace and adjusting
+                    // for multi-line edits that're provided by O#. Right now, we do not support multi-line edits
+                    // (ex. refactoring code actions) however there are certain supported edits which O# is automatically
+                    // formatting for us (ex. FullyQualifiedNamespace) into multiple lines, when it should span a single line.
+                    // This is due to how we render our virtual cs files, with fewer levels of indentation to facilitate
+                    // appropriate error reporting (if we had additional tabs, then the error squigly would appear offset).
+                    //
+                    // The ideal solution for this would do something like:
+                    // https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs#L264
```

Comment:
This will have to be a link to a specific commit otherwise it'll fall out of date (hit `y` when viewing the file to get the perma link)

Reply:
Good catch, updated. Thanks.